src/pages/Applications/Filter/DateInput.tsx

```diff
@@ -0,0 +1,29 @@
+import { ErrorMessage } from "@hookform/error-message";
+import { FieldValues, Path, useFormContext } from "react-hook-form";
+import { FormValues } from "./types";
+
+export default function DateInput<T extends FieldValues>({
+  name,
+}: {
+  name: Path<T>;
+}) {
+  const {
+    register,
+    formState: { errors },
+  } = useFormContext<FormValues>();
```

Comment:
```suggestion
  } = useFormContext<T>();
```
This would make this component generic, so it can be moved to `src/components/form/`

Reply:
fixed